sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -198,6 +198,8 @@ abstract class Optimizer(catalogManager: CatalogManager)
       CheckCartesianProducts) :+
     Batch("RewriteSubquery", Once,
       RewritePredicateSubquery,
+      ConvertToLocalRelation,
+      PropagateEmptyRelation,
```

Comment:
Although this passes PR Builder, this causes consistent failure after merging for all builds.
```
org.apache.spark.sql.hive.HiveSparkSubmitSuite.temporary Hive UDF: define a UDF and use it
org.apache.spark.sql.hive.HiveSparkSubmitSuite.permanent Hive UDF: define a UDF and use it
org.apache.spark.sql.hive.HiveSparkSubmitSuite.permanent Hive UDF: use a already defined permanent function
```

Reply:
This fails at `checkBatchIdempotence` of `RuleExecutor`.
```
[info]   2020-03-30 19:00:21.097 - stderr> 20/03/30 19:00:21 INFO TemporaryHiveUDFTest: Running a simple query on the table.
[info]   2020-03-30 19:00:21.198 - stderr> Exception in thread "main" org.apache.spark.sql.catalyst.errors.package$TreeNodeException:
[info]   2020-03-30 19:00:21.198 - stderr> Once strategy's idempotence is broken for batch RewriteSubquery
[info]   2020-03-30 19:00:21.198 - stderr>  Aggregate [count(1) AS count#15L]       Aggregate [count(1) AS count#15L]
[info]   2020-03-30 19:00:21.198 - stderr>  +- Aggregate [val#5]                    +- Aggregate [val#5]
[info]   2020-03-30 19:00:21.198 - stderr> !   +- Project [val#5]                      +- LocalRelation [val#5]
[info]   2020-03-30 19:00:21.198 - stderr> !      +- LocalRelation [key#4, val#5]
[info]   2020-03-30 19:00:21.198 - stderr>           , tree:
[info]   2020-03-30 19:00:21.198 - stderr> Aggregate [count(1) AS count#15L]
[info]   2020-03-30 19:00:21.198 - stderr> +- Aggregate [val#5]
[info]   2020-03-30 19:00:21.198 - stderr>    +- LocalRelation [val#5]
```

Reply:
I see. We need a separate batch to propagate empty relation after rewriting subquery